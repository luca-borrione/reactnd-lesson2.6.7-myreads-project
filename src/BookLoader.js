import React from 'react';

const BookLoader = () => {
	console.log('>> BookLoader RENDERED');
	return (
		<div className="book-loader">
			<div className="book-page"></div>
			<div className="book-page"></div>
			<div className="book-page"></div>
		</div>
	);

};

export default BookLoader;
