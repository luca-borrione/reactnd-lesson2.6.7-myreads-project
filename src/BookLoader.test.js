import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import BookLoader from './BookLoader';

describe('BookLoader', () => {

	it('renders without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(<BookLoader />, div);
		ReactDOM.unmountComponentAtNode(div);
	});


	it('renders correctly', () => {
		const tree = renderer.create(<BookLoader />)
			.toJSON();

		expect(tree).toMatchSnapshot();
	});

});
