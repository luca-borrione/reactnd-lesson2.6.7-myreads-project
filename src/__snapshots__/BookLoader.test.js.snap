// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookLoader renders correctly 1`] = `
<div
  className="book-loader"
>
  <div
    className="book-page"
  />
  <div
    className="book-page"
  />
  <div
    className="book-page"
  />
</div>
`;
